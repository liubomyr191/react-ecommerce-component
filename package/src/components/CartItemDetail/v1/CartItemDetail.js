import React, { Component } from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { applyTheme } from "../../../utils";

const Detail = styled.div`
  flex: 0 0 fit;

  h3 {
    font-family: ${applyTheme("font_family")};
    letter-spacing: 0.3px;
    line-height: 1;
    margin: 0 0 10px;
  }

  a {
    color: ${applyTheme("color_coolGrey500")};
    text-decoration: none;
    &:focus,
    &:hover {
      color: ${applyTheme("color_coolGrey300")};
    }
  }
`;

const Title = styled.h3`
  font-size: ${applyTheme("font_size_default")};
`;

const Vendor = styled.p`
  color: ${applyTheme("color_black65")};
  display: block;
  font-family: ${applyTheme("font_family")};
  font-size: ${applyTheme("font_size_small")};
  margin: 0;
`;

const Attributes = styled.div`
  margin-bottom: 0.5rem;

  span {
    display: none;

    @media (min-width: 768px) {
      display: inline;
    }
  }
`;

const Attr = styled.p`
  color: ${applyTheme("color_black65")};
  display: inline;
  font-family: ${applyTheme("font_family")};
  font-size: ${applyTheme("font_size_small")};
  margin: 0 0.05rem 0 0;

  &:after {
    content: ",";
  }

  &:last-of-type:after {
    content: "";
  }

  @media (min-width: 768px) {
    display: block;

    &:after {
      content: "";
    }
  }
`;

class CartItemDetail extends Component {
  static propTypes = {
    /**
     * Array of additional attributes of the chosen item.
     */
    attributes: PropTypes.arrayOf(PropTypes.shape({
      /**
         * Attribute label (i.e. "Color").
         */
      label: PropTypes.string,
      /**
         * Attribute value (i.e. "Red").
         */
      value: PropTypes.string
    })),
    /**
     * Product slug of chosen item.
     */
    productSlug: PropTypes.string,
    /**
     * Product title of chosen item.
     */
    title: PropTypes.string
  };

  renderAttributes() {
    let { attributes } = this.props;
    const { value: vendor } = attributes.find((attr) => attr.label === "vendor") || { value: undefined };
    attributes = attributes.filter((attr) => attr.label !== "vendor");
    return (
      <Attributes>
        {vendor ? <Vendor>{vendor}</Vendor> : null}
        {attributes.map(({ label, value }) => (
          <Attr key={label}>
            <span>{label}:</span> {value}
          </Attr>
        ))}
      </Attributes>
    );
  }

  render() {
    const { attributes, productSlug, title } = this.props;
    return (
      <Detail>
        <Title>
          <a href={productSlug}>{title}</a>
        </Title>
        {attributes ? this.renderAttributes() : ""}
      </Detail>
    );
  }
}

export default CartItemDetail;
