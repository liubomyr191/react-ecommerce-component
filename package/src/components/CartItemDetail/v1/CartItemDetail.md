### Overview
Display CartItem title and attributes.

#### Usage

It can be used in Cart or in Mini Cart.

```jsx
const attributes = [{ label: "Color", value: "Red" }, { label: "Size", value: "Medium" }];
<CartItemDetail title="Amazing Product Title" productSlug="/product-slug" attributes={attributes} />
```

##### With Product Vendor
```jsx
const attributes = [{ label: "Color", value: "Red" }, { label: "Size", value: "Medium" }];
<CartItemDetail title="Amazing Product Title" productSlug="/product-slug" productVendor="Patagonia" attributes={attributes} />
```

##### In Mini Cart
```jsx
const attributes = [{ label: "Color", value: "Red" }, { label: "Size", value: "Medium" }];
<CartItemDetail title="Amazing Product Title" productSlug="/product-slug" productVendor="Patagonia" attributes={attributes} isMiniCart />
```
