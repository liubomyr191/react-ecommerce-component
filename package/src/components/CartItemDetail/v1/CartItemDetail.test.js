import React from "react";
import renderer from "react-test-renderer";
import CartItemDetail from "./CartItemDetail";

test("basic snapshot", () => {
  const component = renderer.create(<CartItemDetail />);

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
