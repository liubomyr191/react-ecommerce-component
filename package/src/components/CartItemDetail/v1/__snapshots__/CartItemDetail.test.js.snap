// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic snapshot 1`] = `
.c0 {
  -webkit-flex: 0 0 fit;
  -ms-flex: 0 0 fit;
  flex: 0 0 fit;
}

.c0 h3 {
  font-family: 'Source Sans Pro','Helvetica Neue',Helvetica,sans-serif;
  -webkit-letter-spacing: 0.3px;
  -moz-letter-spacing: 0.3px;
  -ms-letter-spacing: 0.3px;
  letter-spacing: 0.3px;
  line-height: 1;
  margin: 0 0 10px;
}

.c0 a {
  color: #3c3c3c;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 a:focus,
.c0 a:hover {
  color: #5d8ea9;
}

<div
  className="c0"
>
  <h3>
    <a
      href={undefined}
    />
  </h3>
</div>
`;
